### ISSUE TITLE
Sample Project & File Templates

### ISSUE THREAD

**Issue by geekdenz** [Everlaw/nbts#6]
A sample TypeScript project can be under New Project -> Sample ...

Geertjan has implemented a sample project and File Template for TS.

These could reasonably easily be adapted to fit in your project.

Please let me know if/why (not?) I should add this feature here.

**Comment by jeffrey-easyesi** [comment]
I notice the sample project includes generated files (.js and .js.map), should those be there?

**Comment by geekdenz** [comment]
Probably not. They can be generated by the user right?

**Comment by danielschmitz** [comment]
This could help: https://github.com/danielschmitz/simple-netbeans-typescript-project

**Comment by Chris2011** [comment]
I created file templates by my own:

TypeScript Class
```typescript
/**
 *
 * @author ${user}
 */
class ${name} {
    
}
```

TypScript Interface
```typescript
/**
 * 
 * @author ${user}
 */
interface ${name} {
    
}
```

TypeScript Enum
```typescript
/**
 * 
 * @author ${user}
 */
enum ${name} {
    
}
```

I found that useful but I don't want to create a plugin for this, would be nice if you can add it into the NBTS plugin. Here you can find how to register those template files:

https://platform.netbeans.org/tutorials/nbm-filetemplates.html#registering
Names like TypeScript Class, TypeScript Interface, TypeScript Enum, where inspired by the Java Files (Class, Interface, Enum) because of consistency in netbeans. Let me know what you think and if I really have to create a plugin for this, which doesn't make sense, if there is already a plugin which  could handle this. :)

**Comment by Chris2011** [comment]
I know the templates are not that big maybe we can add a module too or imports and so on. if needed.

**Comment by Chris2011** [comment]
Would be nice that such things could implemented into the plugin. ATM, I created the template files which I mentioned, only for me. Better that everyone can have those file templates.

**Comment by geekdenz** [comment]
@Chris2011 
I don't think this should be an additional plugin as you say. Why don't you fork this repository, add the templates and do a pull request with just that change? I think it will be happily accepted. Alternatively, I could do this, but since you are on this already, it is probably more efficient if you just do this. I would +1 this.

**Comment by Chris2011** [comment]
Yeah sounds good. I will do this in the next days if I have time.

**Comment by geekdenz** [comment]
@Chris2011 Did you manage to do it? If not, let me know and I can have a crack at it.

**Comment by Chris2011** [comment]
Sry, mit yet. Im on vacation. Back next week. If you want, do smth. I will contribute to it later or you wait.

**Comment by geekdenz** [comment]
I'll have a look into it. If you supplied all code templates here OK. But if you think more are good, please paste them somewhere.

**Comment by Chris2011** [comment]
i'm back in town, so I can have a look at the next days. Maybe you can prepare smth and I will stick into it :)

**Comment by geekdenz** [comment]
Thanks! Would be great if you can just ask if you are stuck as I'm quite busy at the moment. Sorry, I haven't started on this yet.

**Comment by Chris2011** [comment]
Kk no problem, I will have a look, today or tomorrow.

**Comment by geekdenz** [comment]
@Chris2011 I'm having a look now into this, so you don't need to waste your time on this now if you don't want to. I'll let you know how I go. Got the file. It's
src/netbeanstypescript/resources/tscodetemplates.xml

**Comment by Chris2011** [comment]
This is not wasting time for me @geekdenz, but thx. I only have to find time to do it, like you. But thx for the work I will stick into or tell me what you need. I created 6 file templates.

**Comment by geekdenz** [comment]
I've created some code templates. However, I haven't made this public yet, because I discovered that the templates you seem to be after are slightly different from the ones I added back then. You are after file templates, while I was only familiar with code templates that get replaced when one for example types:

csl<tab>
would expand to
`class name {}`

I will have a look into file templates unless you beat me to it.

**Comment by Chris2011** [comment]
correct my intention was to bring file templates and not code templates to it.

**Comment by geekdenz** [comment]
OK, I figured it out. I should have a pull request in within 48 hours. Please see my fork.
See https://github.com/geekdenz/nbts/tree/code-templates .

**Comment by geekdenz** [comment]
Please only close once I've submitted the pull requests for Interface and Enum as well. Are there others?

**Comment by Chris2011** [comment]
These are my file templates in the folder "TypeScript":

Name: TypeScript Class
Code:
```TypeScript
/**
 *
 * @author ${user}
 */
class ${name} {
    
}
```


Name: TypeScript Interface
Code:
```TypeScript
/**
 * 
 * @author ${user}
 */
interface ${name} {
    
}
```


Name: TypeScript Method Annotation
Code:
```TypeScript
/**
 *
 * @author ${user}
 */
var ${name} = (target: Object, name: String, descriptor: TypedPropertyDescriptor<any>): void => {
    console.log(target);
    console.log(name);
    console.log(descriptor);
};
```


Name: TypeScript Class Annotation
Code:
```TypeScript
/**
 *
 * @author ${user}
 */
var ${name} = <TFunction extends Function>(target: TFunction): void => {
    console.log(target);
};
```


Name: TypeScript Enum
Code:
```TypeScript
/**
 * 
 * @author ${user}
 */
enum ${name} {
    
}
```

**Comment by geekdenz** [comment]
Cool, haven't looked into TS Annotations yet, but will do.

**Comment by geekdenz** [comment]
Sorry, haven't looked into this yet, but I haven't forgotten.

**Comment by geekdenz** [comment]
According to this https://www.typescriptlang.org/docs/handbook/decorators.html decorators are an experimental feature. So, for now, I think it's better not to include code templates for them. Users can add them to their personal templates if they choose to use them.

Thoughts?

**Comment by Chris2011** [comment]
I don't care about that.

**Comment by geekdenz** [comment]
@Chris2011 Do you mean you don't care about decorators, also known as annotations as far as I understand?

**Comment by Chris2011** [comment]
Sry, I meaned I don't care about to add the decorator template or not.

**Comment by geekdenz** [comment]
@jeffrey-easyesi Please accept pull request and close this issue.

**Comment by Rycochet** [comment]
I think there's a specific NetBeans annotation to use to let it use the copyright / comment block set in the NetBeans config, rather than this rather specific "@author" set you're using - can't remember what it is immediately, but might be useful to add :-)

**Comment by geekdenz** [comment]
OK. Might have a look some time soon. Using vim quite happily ATM for editing ts files. But for Java, Netbeans is still a winner. However, I'm happy to contribute here still.

**Comment by Chris2011** [comment]
I know that this is off topic and maybe we should discuss via mail or so but why you have to use vim to editing ts files instead of using NetBeans? Which feature is missing?

**Comment by Chris2011** [comment]
I don't know it anymore, but how was the process? If there is no process atm I can provide the templates for what I have as a PR:

**Comment by Chris2011** [comment]
@jeffrey-easyesi Code templates are not merged yet, please can you accept the PR?
